File: combinatorics/resource/ST/project_5/bai_toan_4/simple_graph/al_to_eal.cpp
#include <iostream>
#include <vector>
#include <sstream>

using namespace std;

// Ham chuyen danh sach ke sang danh sach ke mo rong
vector<vector<pair<int, int>>> adjacencyListToExtended(const vector<vector<int>>& adjList, int n) {
    vector<vector<pair<int, int>>> extendedList(n); // Tao danh sach moi

    for (int i = 0; i < n; ++i) {
        for (int j : adjList[i]) {
            extendedList[i].emplace_back(j, 1); // Them (j, 1) voi trong so mac dinh la 1
        }
    }

    return extendedList;
}

int main() {
    int n;
    cout << "Nhap so dinh: ";
    cin >> n;
    cin.ignore();

    vector<vector<int>> adjList(n);
    cout << "Nhap danh sach ke:\n";

    for (int i = 0; i < n; ++i) {
        cout << "Dinh " << i << ": ";
        string line;
        getline(cin, line);
        stringstream ss(line);
        int x;
        while (ss >> x) {
            adjList[i].push_back(x);
        }
    }

    auto extendedList = adjacencyListToExtended(adjList, n);

    // In danh sach ke mo rong
    cout << "Danh sach ke mo rong (EAL):\n";
    for (int i = 0; i < n; ++i) {
        cout << i << ": ";
        for (const auto& [v, w] : extendedList[i]) {
            cout << "(" << v << ", " << w << ") ";
        }
        cout << endl;
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/simple_graph/al_to_eal.py
# Ham chuyen danh sach ke sang danh sach ke mo rong
def adjacency_list_to_extended(adj_list, n):
    extended_list = [[] for _ in range(n)]  # Tao danh sach moi cho tung dinh

    for i in range(n):
        for j in adj_list[i]:
            extended_list[i].append((j, 1))  # Them (j, 1) vi do thi khong co trong so

    return extended_list

# Nhap du lieu tu ban phim
n = int(input("Nhap so dinh: "))
adj_list = []

print("Nhap danh sach ke:")
for i in range(n):
    line = list(map(int, input(f"Dinh {i}: ").split()))
    adj_list.append(line)

# Chuyen sang EAL va in ra
extended_list = adjacency_list_to_extended(adj_list, n)

print("Danh sach ke mo rong (EAL):")
for i, neighbors in enumerate(extended_list):
    print(f"{i}: {neighbors}")